### PR TITLE
Blob

### PR DESCRIPTION
**Pull request by bddap**
Implements the node side of https://github.com/docknetwork/planning/blob/master/spec/0003-schema.md

### CODE REVIEW

**Comment by bddap** [comment]
@lovesh I don't understand why the `develop` branch is so out of date. It's 19 commits behind master for some reason.

**Comment by bddap** [comment]
Couple naming related changes from the spec.

- `dock::blob::Id` -> `dock::blob::BlobId` because polkadotjs can't handle different types with the same name.
- `BlobModule` -> `BlobStore`
- Querying a blob returns `(Did, Vec<u8>)` instead of `Blob`. BlobId is already known because it is the key to the query.

**Comment by lovesh** [comment]
> @lovesh I don't understand why the `develop` branch is so out of date. It's 19 commits behind master for some reason.

I had started building off an older commit. Will fix this

File: runtime/src/blob.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        BlobMod::new(Origin::signed(ABBA), bl.clone(), sig)
+    }
+
+    fn get_max_blob_size() -> usize {
+        <Test as crate::blob::Trait>::MaxBlobSize::get() as usize
+    }
+
+    #[test]
+    fn add_blob() {
+        if !in_ext() {
```
Comment (lovesh) [inline runtime/src/blob.rs:126]: Same comment on the extra call to the test function as one of the earlier PRs. Let's address this with one of the approaches discussed there.

File: runtime/src/blob.rs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test_common::*;
+
+    fn rando_bytes(len: usize) -> Vec<u8> {
```
Comment (lovesh) [inline runtime/src/blob.rs:99]: Is there a typo in `rando_bytes`

File: runtime/src/test_common.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    Perbill,
+};
+
+pub use frame_support::dispatch::DispatchError;
+pub use rand::random;
+pub use sp_core::sr25519;
+pub use std::iter::once;
+
+pub type RevoMod = crate::revoke::Module<Test>;
+pub type BlobMod = crate::blob::Module<Test>;
```
Comment (lovesh) [inline runtime/src/test_common.rs:21]: `BlobMod` is not used anywhere else so it should not be added here.

**Comment by lovesh** [comment]
@bddap Minor suggestions, otherwise LGTM.

File: runtime/src/blob.rs (hunk 2 above)
Comment (bddap) [reply runtime/src/blob.rs:99]: Rando is an abbreviation of random. You want it changed to rand_bytes or random_bytes?

File: runtime/src/blob.rs (hunk 1 above)
Comment (bddap) [reply runtime/src/blob.rs]: Switched to alternate style in 99a5659120f7f1238a5f93831f3f9de376932e31:

```rust
ext().execute_with(|| {
    ...
});
```

Added new issue https://github.com/docknetwork/dock-substrate/issues/44

File: runtime/src/blob.rs (hunk 2 above)
Comment (bddap) [reply runtime/src/blob.rs]: Changed in 99a5659120f7f1238a5f93831f3f9de376932e31.

`rando_bytes -> random_bytes`

File: runtime/src/test_common.rs (hunk 3 above)
Comment (bddap) [reply runtime/src/test_common.rs]: Addressed in 99a5659120f7f1238a5f93831f3f9de376932e31.

Moved `BlobMod` declaration into blob test module.

**Comment by lovesh** [comment]
> Couple naming related changes from the spec.
> 
> * `dock::blob::Id` -> `dock::blob::BlobId` because polkadotjs can't handle different types with the same name.
> * `BlobModule` -> `BlobStore`
> * Querying a blob returns `(Did, Vec<u8>)` instead of `Blob`. BlobId is already known because it is the key to the query.

Can you update the spec as well?

**Comment by bddap** [comment]
> Can you update the spec as well?

Yep https://github.com/docknetwork/planning/pull/4

**Pull request merged**